docs/cn/built_in_function_develop_guide.md

```diff
@@ -0,0 +1,230 @@
+# OpenMLDB SQL 内置函数开发指南
+
+## 背景
+
+​	OpenMLDB内置了上百个SQL函数，以供数据科学家作数据分析和特征抽取。目前，我们提供聚合类函数(Aggregate Function)如 `SUM`, `AVG`, `MAX`, `MIN`, `COUNT`来支持全表聚合和窗口聚合。同时，我们还提供了单行函数(Scalar Function)如`ABS`, `SIN`, `COS`, `DATE`, `YEAR`等支持单行数据处理。
+
+​	本文是SQL内置函数的开发入门指南，旨在指引开发者快速掌握基础的内置函数开发方法。我们诚挚欢迎更多的开发者加入社区，帮助我们扩展和开发内置函数集。
+
+## 函数分类
+
+OpenMLDB将函数分类两大类：聚合类函数和单行处理函数:
+
+- 聚合函数((Aggregate Function)对数据集（如一列数据）执行计算，并返回单个值。
+- 单行函数（Scalar Function)对若干数据执行计算，并返回单个值。根据函数功能，又可分为
+  - 数学函数
+  - 逻辑函数
+  - 日期函数
+  - 字符串函数
+
+本文将介绍单行函数的开发，暂不深入更为复杂的聚合函数开发细节。
+
+## 开发和注册内置函数
+
+`ExternalFuncRegistryHelper` is OpenMLDB的内置函数开发最重要的辅助类。它提供一套API来配置函数，并将配置好的函数注册到相应的函数库中。完成函数注册后，可以在SQL查询语句中调用函数。本章节主要介绍内置函数的注册和配置的基本步骤和并提供若干示范。
+
+### ExternalFuncRegistryHelper API
+
+OpenMLDB内置的函数都应该注册到默认函数库`DefaultUdfLibrary`(src/udf/default_udf_library.cc)中。`DefaultUdfLibrary`提供`RegisterExternal` 接口来创建一个函数注册助手。它负责配置函数的参数，配置返回值了行，并注册函数到默认函数库中。
+
+```c++
+RegisterExternal(function_name)
+  .args<arg_type, ...>(built_in_fn_pointer)
+  .return_by_arg(bool_value)
+  .returns<return_type>
+  .doc(documentation)
+```
+
+- `RegisterExternal(function_name)`: 可以创建一个`ExternalFuncRegistryHelper`对象，初始化函数的注册名。SQL使用这个名字来调用函数。
+- `built_in_fn_pointer`: 被注册的C++函数指针
+- `args<arg_type,...>`: 配置函数参数类型
```

Comment:
It is better to describe the type of function parameter. ex: the difference between `hybridse::codec::StringRef`  and  `std::string` , and how to select

Reply:
I will add a section like:
### How to Implement a built-in function
here,
I will describe the type mapping from OpenMLDB type to c++ type.

Reply:
done